app/kuma-dp/pkg/dataplane/envoy/envoy.go

```diff
@@ -51,6 +97,13 @@ func (e *Envoy) Run(stop <-chan struct{}) error {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
+	binaryPathConfig := e.opts.Config.DataplaneRuntime.BinaryPath
+	resolvedPath, err := lookupEnvoyPath(binaryPathConfig)
+	if err != nil {
+		runLog.Error(err, "Envoy binary not found; make sure it is in PATH or in the same directory as "+os.Args[0])
+		return nil
```

Comment:
At the moment, if `envoy` binary is not found, `konvoy-dp` exists with status code `0` and prints nothing on the console.

You should return the error instead of logging it.